csharp/src/Apache.Arrow/Schema.cs

```diff
@@ -63,5 +69,38 @@ public int GetFieldIndex(string name, StringComparer comparer = default)
             return _fields.IndexOf(
                 _fields.Single(x => comparer.Compare(x.Name, name) == 0));
         }
+
+        public void RemoveField(int fieldIndex)
```

Comment:
Are `Schema` objects supposed to be immutable?

Reply:
Interesting, the spec doesn't actually say anything about immutability: https://arrow.apache.org/docs/format/Metadata.html

The C++ lib first creates a new `Schema` on the heap and then returns a new `Table` object(also allocated on the heap) as an `out` parameter. I wonder why it does that. In any case, I modified this to be similar to the C++ API